src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -38,11 +38,22 @@ void GcsResourceManager::ConsumeSyncMessage(
   // delegate the work to the main thread for thread safety.
   // Ideally, all public api in GcsResourceManager need to be put into this
   // io context for thread safety.
+
+  // We should only expect RESOURCE_VIEW messages from the syncer for GcsResourceManager.
+  // See GcsServer::InitRaySyncer for syncer registration.
+  RAY_CHECK(message->message_type() == syncer::MessageType::RESOURCE_VIEW);
```

Comment:
RAY_CHECK is a bit aggressive. how about ignore the message if it's not RESOURCE_VIEW

Reply:
I'm not sure, but it shouldn't happen unless there is bug in ray syncer.
We should at least print RAY_ERROR if you are not comfortable with this and need a fix asap.

Reply:
yah, i think this message type assertion should be rather safe (in cases when ray works). Otherwise, it will be a serious bug as Yi mentioned.